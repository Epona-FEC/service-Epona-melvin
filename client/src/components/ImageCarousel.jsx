import React from 'react';
import PropTypes from 'prop-types';

const ImageCarousel = ({
  photos, slideInNextImages, slideInPrevImages, styles,
}) => (
  <div className='carousel-container'>
    <button
      type='button'
      className="review-prev-button"
      onClick={slideInPrevImages}
    >
      &lt;
    </button>
    <button
      type='button'
      className="review-next-button"
      onClick={slideInNextImages}
    >
      &gt;
    </button>
    <div
      className='carousel-slide'
      style={styles}
    >
      {photos.map((photo) => <img className='review-carousel-photo' src={photo.url} alt={photo.description} />)}
    </div>

  </div>
);


ImageCarousel.propTypes = {
  photos: PropTypes.arrayOf(
    PropTypes.shape({
      url: PropTypes.string,
      description: PropTypes.string,
    }),
  ),
  slideInPrevImages: PropTypes.func,
  slideInNextImages: PropTypes.func,
  styles: PropTypes.shape({
    transform: PropTypes.string,
  }),
};

ImageCarousel.defaultProps = {
  photos: [
    {
      url: 'https://source.unsplash.com/random/200x200',
      description: 'Purchase of a customer',
    },
    {
      url: 'https://source.unsplash.com/random/200x200',
      description: 'Purchase of a customer',
    },
  ],
  slideInPrevImages: () => null,
  slideInNextImages: () => null,
  styles: {
    transform: 'translateX(0)',
  },
};


export default ImageCarousel;
