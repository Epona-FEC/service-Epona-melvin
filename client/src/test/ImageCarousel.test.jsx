/* eslint-disable no-undef */
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import ImageCarousel from '../components/ImageCarousel.jsx';

Enzyme.configure({ adapter: new Adapter() });

describe('ImageCarousel', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<ImageCarousel />);
  });

  it('Should shallow', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Function should be invoked, when prevButton is called', () => {
    let counter = 0;
    const func = () => { counter += 1; };
    wrapper.setProps({ slideInPrevImages: func });
    wrapper.find('.review-prev-button').simulate('click');
    expect(counter).toBe(1);
  });

  it('Function should be invoked, when nextButton is called', () => {
    let counter = 0;
    const func = () => { counter += 1; };
    wrapper.setProps({ slideInNextImages: func });
    wrapper.find('.review-next-button').simulate('click');
    expect(counter).toBe(1);
  });
});
