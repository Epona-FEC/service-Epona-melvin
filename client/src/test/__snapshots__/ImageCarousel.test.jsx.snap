// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageCarousel Should shallow 1`] = `
<div
  className="carousel-container"
>
  <button
    className="review-prev-button"
    onClick={[Function]}
    type="button"
  >
    &lt;
  </button>
  <button
    className="review-next-button"
    onClick={[Function]}
    type="button"
  >
    &gt;
  </button>
  <div
    className="carousel-slide"
    style={
      Object {
        "transform": "translateX(0)",
      }
    }
  >
    <img
      alt="Purchase of a customer"
      className="review-carousel-photo"
      src="https://source.unsplash.com/random/200x200"
    />
    <img
      alt="Purchase of a customer"
      className="review-carousel-photo"
      src="https://source.unsplash.com/random/200x200"
    />
  </div>
</div>
`;
